app/editor/src/upload-context/UploadContextWrapper.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { toast } from 'react-toastify';
+import { useStorage } from 'store/hooks';
+
+import { IUploadFile, IUploadManager } from './interfaces';
+import { UploadContext, UploadFilesContext } from './UploadContext';
+
+export interface IUploadContextWrapperProps {
+  children: React.ReactElement;
+}
+
+export const UploadContextWrapper: React.FC<IUploadContextWrapperProps> = ({ children }) => {
+  const [file, setFile] = React.useState<IUploadFile>();
+  const storage = useStorage();
+
+  const toastId = React.useRef<any>(null);
+
+  const upload = React.useCallback(
+    (file: File) => {
+      const data = new FormData();
+      data.append('file', file);
+
+      setFile({
+        name: file.name,
+        status: 'pending',
+        loaded: 0,
+        total: file.size,
+      });
+      storage
```

Comment:
For now this will be okay, but we can't have the multi-upload tightly coupled with the storage hook.